cli/src/commands/git/push.rs

```diff
@@ -746,6 +767,65 @@ fn classify_bookmark_update(
     }
 }
 
+/// Creates a bookmark for a single `--named` argument and returns its name
+///
+/// The logic is not identical to that of `jj bookmark create` since we need to
+/// make sure the new bookmark is safe to push.
+fn create_explicitly_named_bookmarks(
+    ui: &mut Ui,
+    tx: &mut WorkspaceCommandTransaction<'_>,
+    name_revision: &String,
+) -> Result<RefNameBuf, CommandError> {
+    let hint = "For example, `--named myfeature=@` is valid syntax";
+    let Some((name_str, revision_str)) = name_revision.split_once('=') else {
+        return Err(cli_error(format!(
+            "Argument '{name_revision}' must include '=' and have the form NAME=REVISION"
+        ))
+        .hinted(hint));
+    };
+    // Revisions are lenient towards spaces, so let's make the name lenient as well
+    // Git does not allow bookmark names that include spaces anyway.
+    let name_str = name_str.trim();
+    if name_str.is_empty() || revision_str.is_empty() {
+        return Err(cli_error(format!(
+            "Argument '{name_revision}' must have the form NAME=REVISION, with both NAME and \
+             REVISION non-empty."
+        ))
+        .hinted(hint));
+    }
+    let name = parse_bookmark_name(name_str).map_err(|err| {
+        cli_error(format!(
+            "Could not parse '{name_str}' as a bookmark name: {err}",
+        ))
```

Comment:
nit: `CommandError::with_message(CommandErrorKind::Cli, format!(..), err)` or add `cli_error_with_message()`?

Reply:
Worked great, thanks!

Reply:
We could also later separate much of this function into a clap `value_parser`. I don't see a compelling reason to do it right now, though I *am* hoping to implement command-line completion for the revision part of NAME=REVISION.

Reply:
Yeah, `KEY=VALUE` splitting could be done by clap. We'll have to parse `VALUE` as revset expression separately, though.